x/upgrade/abci_test.go

```diff
@@ -49,104 +49,96 @@ func setupTest(height int64, skip []int64) TestSuite {
 	return s
 }
 
-// this is a temporary way to unify TestSuite.SetupTest and setupTest
-// can be merged into setupTest when TestSuite goes away
-func setupTestSuiteInPlace(s *TestSuite, height int64, skip []int64) {
-
-}
-
 func TestRequireName(t *testing.T) {
 	s := setupTest(10, []int64{})
 
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{}})
-	require.NotNil(t,err)
-	require.Equal(t,sdk.CodeUnknownRequest, err.Code())
+	require.NotNil(t, err)
+	require.Equal(t, sdk.CodeUnknownRequest, err.Code())
 }
 
 func TestRequireFutureTime(t *testing.T) {
 	s := setupTest(10, []int64{})
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Time: s.ctx.BlockHeader().Time}})
-	require.NotNil(t,err)
-	require.Equal(t,sdk.CodeUnknownRequest, err.Code())
+	require.NotNil(t, err)
+	require.Equal(t, sdk.CodeUnknownRequest, err.Code())
 }
 
 func TestRequireFutureBlock(t *testing.T) {
 	s := setupTest(10, []int64{})
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Height: s.ctx.BlockHeight()}})
-	require.NotNil(t,err)
-	require.Equal(t,sdk.CodeUnknownRequest, err.Code())
+	require.NotNil(t, err)
+	require.Equal(t, sdk.CodeUnknownRequest, err.Code())
 }
 
 func TestCantSetBothTimeAndHeight(t *testing.T) {
 	s := setupTest(10, []int64{})
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Time: time.Now(), Height: s.ctx.BlockHeight() + 1}})
-	require.NotNil(t,err)
-	require.Equal(t,sdk.CodeUnknownRequest, err.Code())
+	require.NotNil(t, err)
+	require.Equal(t, sdk.CodeUnknownRequest, err.Code())
 }
 
 func TestDoTimeUpgrade(t *testing.T) {
 	s := setupTest(10, []int64{})
 	t.Log("Verify can schedule an upgrade")
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Time: time.Now()}})
-	require.Nil(t,err)
+	require.Nil(t, err)
 
-	VerifyDoUpgrade(t)
+	s.VerifyDoUpgrade(t)
 }
 
 func TestDoHeightUpgrade(t *testing.T) {
 	s := setupTest(10, []int64{})
 	t.Log("Verify can schedule an upgrade")
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Height: s.ctx.BlockHeight() + 1}})
-	require.Nil(t,err)
+	require.Nil(t, err)
 
-	VerifyDoUpgrade(t)
+	s.VerifyDoUpgrade(t)
 }
 
 func TestCanOverwriteScheduleUpgrade(t *testing.T) {
 	s := setupTest(10, []int64{})
 	t.Log("Can overwrite plan")
 	err := s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "bad_test", Height: s.ctx.BlockHeight() + 10}})
-	require.Nil(t,err)
+	require.Nil(t, err)
 	err = s.handler(s.ctx, upgrade.SoftwareUpgradeProposal{Title: "prop", Plan: upgrade.Plan{Name: "test", Height: s.ctx.BlockHeight() + 1}})
-	require.Nil(t,err)
+	require.Nil(t, err)
 
-	VerifyDoUpgrade(t)
+	s.VerifyDoUpgrade(t)
```

Comment:
Hey, that is a nice solution. Still keep the methods on `TestSuite` struct (as we need the keeper, etc), but no longer depend on the `suite.Suite` object exposing `.T()`. Nice fix up